### PR TITLE
fix satrn for ORT

### PR DESCRIPTION
**Pull request by AllentDan**
Fix issue #735

### CODE REVIEW

File: mmdeploy/codebase/mmocr/models/text_recognition/satrn_decoder.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    """Rewrite `forward` of ScaledDotProductAttention for default backend.
+
+    Replace `attn.masked_fill(mask == 0, float('-inf'))` with
+    `attn.masked_fill(mask == 0, -1e34)`.
+    """
+    attn = torch.matmul(q / self.temperature, k.transpose(2, 3))
+
+    if mask is not None:
+        # use float('-inf') may raise NAN for onnx
+        attn = attn.masked_fill(mask == 0, -1e34)
```
Comment (grimoire) [inline mmdeploy/codebase/mmocr/models/text_recognition/satrn_decoder.py:22]: Can we rewrite `masked_fill`?

File: mmdeploy/codebase/mmocr/models/text_recognition/satrn_decoder.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import torch.nn.functional as F
+
+from mmdeploy.core import FUNCTION_REWRITER
+from mmdeploy.utils.constants import Backend
+
+
+@FUNCTION_REWRITER.register_rewriter(
+    func_name='mmocr.models.common.ScaledDotProductAttention.forward',
+    backend=Backend.ONNXRUNTIME.value)
+def scaled_dot_product_attention__forward(ctx, self, q, k, v, mask=None):
```
Comment (RunningLeon) [inline mmdeploy/codebase/mmocr/models/text_recognition/satrn_decoder.py:12]: may add `__onnxruntime` in the function name

Comment (RunningLeon) [reply mmdeploy/codebase/mmocr/models/text_recognition/satrn_decoder.py:12]: could add unit test for this rewriting if possible.

**Review by grimoire — APPROVED** [review]
LGTM

**Review by RunningLeon — APPROVED** [review]
LGTM

**Pull request merged**